### PR TITLE
[#153] Add render legal moves preview

### PR DESCRIPTION
**Pull request by benshumaker**
Closes: #153

### CODE REVIEW

File: frontend/packages/web/src/components/features/BoardDisplay/Square.tsx [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
+    const moves = Object.values(item.available_moves).flat();
+    if (moves.some((move) => move.rank == i && move.file == j)) {
+      return true;
+    }
+    return false;
+  };
+
+  const handleOnDrop = (item: Piece) => {
+    if (isValidMove(item)) {
+      console.log('Valid Move');
```
Comment (izakfr) [inline frontend/packages/web/src/components/features/BoardDisplay/Square.tsx:27]: Move console log please. Or at least add a TODO.

File: frontend/packages/web/src/components/features/BoardDisplay/Square.tsx [hunk 2]
```diff
@@ -1,22 +1,63 @@
...
+      return 'brightness(0.6)';
+    }
+    return 'none';
+  };
+
   const [{ isOver, canDrop }, drop] = useDrop(
     () => ({
       accept: 'piece',
-      canDrop: () => true,
+      canDrop: initializeDrag, // Solve to show only legal moves
```
Comment (izakfr) [inline frontend/packages/web/src/components/features/BoardDisplay/Square.tsx:41]: Remove comment please.

File: frontend/packages/web/src/models/game.ts [hunk 3]
```diff
@@ -1,6 +1,14 @@
 export type Game = {
   id: string;
   clocks: { [playerId: string]: number };
+  gameboard_state: {
```
Comment (izakfr) [inline frontend/packages/web/src/models/game.ts:4]: Nice type update!

File: frontend/packages/web/src/store/atoms/rooms.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { atom } from 'recoil';
-import { Player, Room, Game } from '../../models';
+import { Player, Room, Game, Piece } from '../../models';
```
Comment (izakfr) [inline frontend/packages/web/src/store/atoms/rooms.ts:2]: Can you change this file's name to state.ts?

**Review by izakfr — CHANGES_REQUESTED** [review]

File: frontend/packages/web/src/components/features/BoardDisplay/Piece.tsx [hunk 5]
```diff
@@ -1,28 +1,33 @@
...
-    ];
-
-  const image = new Image();
-  image.src = pieceImageSource;
-  dragPreview(image);
+    pieceImages[piece.piece_type as PieceType][piece.color as keyof PieceColors];
 
   return (
-    <div ref={drag}>
+    <div ref={drag} style={{ opacity: isDragging ? 0 : 1 }}>
```
[the remark this answers is not in the document]
Comment (benshumaker) [reply frontend/packages/web/src/components/features/BoardDisplay/Piece.tsx:30]: Yes

File: frontend/packages/web/src/components/features/BoardDisplay/Square.tsx (hunk 2 above)
Comment (benshumaker) [reply frontend/packages/web/src/components/features/BoardDisplay/Square.tsx:41]: Agree

File: frontend/packages/web/src/components/features/BoardDisplay/Square.tsx (hunk 1 above)
Comment (benshumaker) [reply frontend/packages/web/src/components/features/BoardDisplay/Square.tsx:27]: I'll add a TODO comment

**Review by izakfr — APPROVED** [review]
Thanks!

**Pull request merged**